library/std/src/sys/unix/process/process_unix.rs

```diff
@@ -719,14 +719,15 @@ impl Process {
     pub fn kill(&mut self) -> io::Result<()> {
         // If we've already waited on this process then the pid can be recycled
         // and used for another process, and we probably shouldn't be killing
-        // random processes, so just return an error.
+        // random processes, so return Ok because the process has exited already.
         if self.status.is_some() {
-            Err(io::const_io_error!(
-                ErrorKind::InvalidInput,
-                "invalid argument: can't kill an exited process",
-            ))
+            Ok(())
         } else {
-            cvt(unsafe { libc::kill(self.pid, libc::SIGKILL) }).map(drop)
+            match cvt(unsafe { libc::kill(self.pid, libc::SIGKILL) }).map(drop) {
+                // If the process can't be found or is already a zombie, return `Ok`.
+                Err(e) if e.raw_os_error() == Some(libc::ESRCH) => Ok(()),
```

Comment:
This seems to be incorrect: it is fine to send `kill` to a zombie process, it is no-nop. And `ESRCH` actually indicates a serious problem: someone else has waited our process, so we should fail.